### ISSUE TITLE
Non-JSON response on some RPC methods when the request returns an error (same as #7006)

### ISSUE THREAD

**Issue by Denis-score** [ledgerwatch/erigon#7033]
Same as ticket #7006 after update to version 2.40.0 who should fix this, we still have invalid json response from certains call like : 

Call : 
```
{
	"jsonrpc":"2.0",
	"method":"debug_traceTransaction",
	"params":[
		"0x58f35cae80d0d50fc6fb8a6ffd3ae2a6e1180366cabe0772a5bd91d7466ff686", { "tracer": "callTracer"}
	],
	"id":"1"
}
```

Response :  

```
{
    "jsonrpc": "2.0",
    "id": "1",
    "result":nullnull,
    "error": {
        "code": -32000,
        "message": "EOF"
    }
}
```

**Comment by AskAlexSharov** [comment]
@hexoscott plz help to add this case to unit-test, thanks.

**Comment by jcchevalier-ledger** [comment]
Same on 2.40.1

**Comment by hexoscott** [comment]
Hi @jcchevalier-ledger, this issue should now be solved by #7104.  I'm going to close this issue down, but feel free to re-open if the problem persists when this finds its way into a release.
